### ISSUE TITLE
Unable to save after going to sleep.

### ISSUE THREAD

**Issue by S0ndz** [Esca-MMC/FarmTypeManager#24]
Hi, having an issue in a year 1 save:

https://smapi.io/log/883ef20f2f4144699eccdcf1caed7a97

Going to bed and trying to save will just permanently keep me in the saving screen.

**Comment by Esca-MMC** [comment]
Sorry about the delay. If this is still an issue, please post another log, since this one has expired.

There are also some bug fixes included in the upcoming FTM v1.16.1 that might prevent this.

**Comment by Creedowl** [comment]
I encounted the same issue. The error is here:
```[22:31:21 TRACE game] saveTask failed with an exception
[22:31:21 ERROR game] InvalidOperationException: There was an error generating the XML document.
 ---> InvalidOperationException: The type FarmTypeManager.Monsters.SerpentFTM was not expected. Use the XmlInclude or SoapInclude attribute to specify types that are not known statically.
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriterSaveGame.Write100_NPC(String n, String ns, NPC o, Boolean isNullable, Boolean needType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriterSaveGame.Write188_GameLocation(String n, String ns, GameLocation o, Boolean isNullable, Boolean needType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriterSaveGame.Write253_SaveGame(String n, String ns, SaveGame o, Boolean isNullable, Boolean needType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriterSaveGame.Write254_SaveGame(Object o)
   --- End of inner exception stack trace ---
   at System.Xml.Serialization.XmlSerializer.Serialize(XmlWriter xmlWriter, Object o, XmlSerializerNamespaces namespaces, String encodingStyle, String id)
   at SpaceCore.Patches.SaveGamePatcher.SerializeProxy(XmlSerializer serializer, XmlWriter origWriter, Object obj) in C:\Programming\StardewValley\ModSource\SpaceCore\Patches\SaveGamePatcher.cs:line 401
   at StardewValley.SaveGame+<getSaveEnumerator>d__90.MoveNext_PatchedBy<spacechase0.SpaceCore>(<getSaveEnumerator>d__90 this)
   at StardewValley.SaveGame.<>c__DisplayClass88_0.<Save>b__0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
[22:31:21 ERROR SMAPI] An error occurred in the overridden update loop: InvalidOperationException: There was an error generating the XML document.
 ---> InvalidOperationException: The type FarmTypeManager.Monsters.SerpentFTM was not expected. Use the XmlInclude or SoapInclude attribute to specify types that are not known statically.
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriterSaveGame.Write100_NPC(String n, String ns, NPC o, Boolean isNullable, Boolean needType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriterSaveGame.Write188_GameLocation(String n, String ns, GameLocation o, Boolean isNullable, Boolean needType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriterSaveGame.Write253_SaveGame(String n, String ns, SaveGame o, Boolean isNullable, Boolean needType)
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriterSaveGame.Write254_SaveGame(Object o)
   --- End of inner exception stack trace ---
   at StardewValley.SaveGame.Save()+MoveNext() in stardewvalley\Farmer\Farmer\SaveGame.cs:line 416
   at StardewValley.Menus.SaveGameMenu.update_PatchedBy<spacechase0.SpaceCore>(SaveGameMenu this, GameTime time)
   at StardewValley.Game1._update(GameTime gameTime)
   at StardewValley.Game1.Update(GameTime gameTime)
   at StardewModdingAPI.Framework.SCore.OnPlayerInstanceUpdating(SGame instance, GameTime gameTime, Action runUpdate)
```
and here is the log: https://smapi.io/log/9d5ec140e8b04a858675ce96f6f8cf4b?format=RawView

I've already install the latest version of FTM v1.6.1, and I found if I save the game with [save anywhere  v.3.2.7](https://www.nexusmods.com/stardewvalley/mods/8386), quit the game and restart it, then load the save, it will work. I'll enter game, and it will save sucessfully and go to the next if I go to bed. Only the time may back to 6:00 AM rather than when I save it with  save anywhere.

**Comment by Esca-MMC** [comment]
Thank you for the log. I'll see if there's anything I can do. This might need to be fixed by Save Anywhere's developers instead, though. You should report it here as well: [https://www.nexusmods.com/stardewvalley/mods/8386?tab=bugs](https://www.nexusmods.com/stardewvalley/mods/8386?tab=bugs)

**Comment by helenaseb** [comment]
Hi, I am also having the same problem. Is there a solution? Thank you

**Comment by Esca-MMC** [comment]
@helenaseb: Avoid using Save Anywhere and FTM together until one of the mods updates and mentions fixing it. They currently aren't fully compatible, so for some players, saving in the middle of the day will cause FTM errors later.

**Comment by Esca-MMC** [comment]
This should hopefully be fixed in FTM v1.16.2. Save Anywhere's developer is still working on updates too.
